sg14-error-codes.html

```diff
@@ -664,6 +714,10 @@
     It is impossible to manipulate any <code>error_category</code> instance constexprly, because <code>error_category</code>
     is not a literal type: it has a virtual destructor, whereas literal types require trivial destructors. This is especially
     unfortunate because <code>error_category</code> does not need polymorphic destruction.
+    
+    <!-- The virtual destructor is to allow derived classes to free caches or tables of message lookups, indeed any Windows
+    implementation of system_category would need this as it queries the Windows kernel and stores retrieved strings into
+    a hash table -->
```

Comment:
This is a reason to provide a destructor in the derived class. It is not a reason to make the base class's destructor *virtual*. If I'm wrong, please provide more details, because I'm not seeing the relevance of your `system_category` example yet.

If each category is a singleton with global lifetime, it's not like I'm ever going to be calling `delete` on an `error_category*` and requiring *virtual dispatch* to find the right destructor. What's actually going to happen is that somewhere in the global cleanup code there's going to be a call to `~system_category`, which will be compiled into a non-virtual dispatch because it knows the exact type of the singleton variable that was *actually constructed.*

If we're not assuming that categories are singletons, then my logic might not apply; but then I would have basically no idea what assumptions we're still allowing ourselves to make.

Reply:
That's a reasonable point, but equally I don't personally think it's actually a big thing in practice. Derived classes can simply mark their destructor as `final`. Runtime problem fixed. The remaining issue is lack of constexpr construction, but you need a whole new system_error2 for that. Though see new topic on sg14 that I'll post shortly.